docs/guide/profiling-test-performance.md

```diff
@@ -0,0 +1,81 @@
+# Profiling Test Performance
+
+When you run Vitest it reports multiple time metrics of your tests:
+
+> ```bash
+> RUN  v2.1.1 /x/vitest/examples/profiling
+>
+> ✓ test/prime-number.test.ts (1) 4517ms
+>   ✓ generate prime number 4517ms
+>
+> Test Files  1 passed (1)
+>      Tests  1 passed (1)
+>   Start at  09:32:53
+>   Duration  4.80s (transform 44ms, setup 0ms, collect 35ms, tests 4.52s, environment 0ms, prepare 81ms)
+>   # Time metrics ^^
+> ```
+
+- Transform: How much time was spent transforming the files.
+- Setup: Time spent for running the [`globalSetup`](/config/#globalsetup) files.
+- Collect: Time spent for colleting all tests in the test files. This includes the time it took to import all file dependencies.
+- Tests: Time spent for actually running the test cases.
+- Environment: Time spent for setting up the test environment, for example JSDOM.
+- Prepare: ???
+
+## Test runner
+
+In cases where your test execution time is high, you can generate a profile of the test runner. Define `poolOptions.<pool>.execArgv` in your Vitest configuration:
+
+::: code-group
+```ts [Forks]
+import { defineConfig } from 'vitest/config'
+
+export default defineConfig({
+  test: {
+    pool: 'forks',
+    poolOptions: {
+      forks: {
+        // TODO, requires https://github.com/nodejs/node/issues/55094
+        execArgv: ['--cpu-prof', '--cpu-prof-dir=test-runner-profile'],
+
+        // To generate a single profile
+        singleFork: true,
+      },
+    },
+  },
+})
+```
+```ts [Threads]
+import { defineConfig } from 'vitest/config'
+
+export default defineConfig({
+  test: {
+    pool: 'threads',
+    poolOptions: {
+      threads: {
+        execArgv: ['--cpu-prof', '--cpu-prof-dir=test-runner-profile'],
+
+        // To generate a single profile
+        singleThread: true,
+      },
+    },
+  },
+})
+```
+:::
+
+After the tests have run there should be a `test-runner-profile/*.cpuprofile` file generated. You can inspect the contents for example with Chrome Dev Tools: https://developer.chrome.com/docs/devtools/performance/nodejs#analyze.
+
+## Main thread
+
+Profiling main thread is useful for debugging [`globalSetup`](/config/#globalsetup) files and Vitest itself. To do this, you'll need to pass arguments to the Node process that runs Vitest.
+
+```bash
+$ node --cpu-prof --cpu-prof-dir=main-profile ./node_modules/vitest/vitest.mjs
+```
+
+After the tests have run there should be a `main-profile/*.001.cpuprofile` file generated. You can inspect the contents for example with Chrome Dev Tools: https://developer.chrome.com/docs/devtools/performance/nodejs#analyze.
+
+## File transform
+
+In cases where your test transform and collection time is high, you can ...
```

Comment:
Is the best option to describe usage of `VITE_NODE_DEBUG_DUMP`?

https://github.com/vitest-dev/vitest/blob/87232420ebe0d5b6052529c0043ae4981590caf1/packages/vite-node/README.md?plain=1#L132-L136

Reply:
I often use `VITE_NODE_DEBUG_DUMP` or `DEBUG=vite-node:*` to see which files are transformed but is it useful for users to improve performance by themselves?

Reply:
I think it would be great to show how to debug cases where file transforms are taking too long. Seeing which files were transformed can reveal issues with barrel files, for example in here: https://github.com/vitest-dev/vitest/issues/1861#issuecomment-1416072983

Reply:
Indeed, yeah barrel files are something actionable for users. I guess some performance tips will overlap with Vite's doc https://vitejs.dev/guide/performance.html

Btw, I just remembered Vitest UI can shows transform in module graph tab.

Reply:
The module graph on UI looks great! I'll use it to highlight the impact of barrel file import here.